src/provider.rs

```diff
@@ -189,7 +200,18 @@ impl StatusLineElement {
             }
             Some(value)
         } else {
-            None
+            // empty tags (no metadata found): when hide_empty is false,
+            // we will return an empty string to make sure all metatags are
+            // added to the status line. This way, we can postprocess the status line
+            // being sure that parameters stay at their position.
+
+            // return None if we do not want to use it further
+            if line_settings.hide_empty {
```

Comment:
I'd move this code out to the caller:
 1. its the only place where `line_settings` is required, and
 2. It doesn't match what the function is supposed to do.

Reply:
True (and fixed). But I think we're going to remove this anyway because this leads in to an impasse.